### PR TITLE
EDX-4381 Add mqtt5 Publish function

### PR DESCRIPTION
**Pull request by jinlinGuan**

### CODE REVIEW

File: pkg/mqtt5/interface.go [hunk 1]
```diff
@@ -26,4 +27,8 @@ type MessageClient interface {
 
 	// Unsubscribe to unsubscribe from the specified topics.
 	Unsubscribe(topics []string) error
+
+	// Publish is to send message to the message bus
+	// the message contains data payload to send to the message queue
+	Publish(topic string, message models.MessageEnvelope) error
```
Comment (cloudxxx8) [inline pkg/mqtt5/interface.go:33]: use the errors.Error interface for all the error type in this interface
maybe open another Jira ticket and modify in the next PR

**Review by cloudxxx8 — CHANGES_REQUESTED** [review]

File: pkg/mqtt5/client.go [hunk 2]
```diff
@@ -220,9 +223,11 @@ func (c *Mqtt5Client) Unsubscribe(topics []string) error {
 	defer c.mutex.Unlock()
 
 	ua, err := c.mqtt5Client.Unsubscribe(c.ctx, &paho.Unsubscribe{Topics: topics})
-	for _, code := range ua.Reasons {
-		if code != 0 {
-			c.logger.Debugf("Received an MQTT 5 error code: 0x%02x - %s", code, ua.Properties.ReasonString)
+	if ua != nil {
+		for _, code := range ua.Reasons {
+			if code != 0 {
```
[the remark this answers is not in the document]
Comment (jinlinGuan) [reply pkg/mqtt5/client.go:228]: No, only subscription results can return Qos values

File: pkg/mqtt5/client.go [hunk 3]
```diff
@@ -235,6 +240,33 @@ func (c *Mqtt5Client) Unsubscribe(topics []string) error {
...
+
+	if pa != nil && pa.ReasonCode != 0 {
+		c.logger.Debugf("Received an MQTT 5 error code: 0x%02x - %s", pa.ReasonCode, pa.Properties.ReasonString)
+	}
+	if err != nil {
+		c.logger.Errorf("Failed to send message to %s", topic)
+		return err
+	}
+
+	c.logger.Infof("Message is sent to %s", topic)
```
[the remark this answers is not in the document]
Comment (jinlinGuan) [reply pkg/mqtt5/client.go]: added

File: pkg/mqtt5/interface.go (hunk 1 above)
Comment (jinlinGuan) [reply pkg/mqtt5/interface.go]: updated

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Pull request merged**